crossing-repair.lic

```diff
@@ -10,18 +10,72 @@ class CrossingRepair
+        { name: 'debug', regex: /debug/i, optional: true, description: 'Enable debug mode' }
+      ]
+    ]
+    args = parse_args(arg_definitions)
 
-    start_script('performance', ['noclean']) unless settings.instrument
+    $debug_mode_crossing_repair = args.debug || UserVars.crossing_repair_debug
+
+    town = DRC.get_town_name(args.town)
+    settings = get_settings
+
+    # If town argument is not recognized, choosing to exit the script with error
+    # instead of continue with the character's hometown because if the player
+    # wanted to repair in a specific town then they might be far away from their
+    # hometown and not want the script to suddenly rush them off,
+    # especially if it was a simple typo or the player suffers social outrage.
+    if args.town && !town
+      DRC.message("Could not identify town for argument: #{args.town}")
+      DRC.message("To avoid ambiguity, please use the town's full name: https://elanthipedia.play.net/Category:Cities")
+      exit
+    end
+
+    # If the user has specified a repair town via script call argument, set that here.
+    settings['hometown'] = town if town
+
+    @repair_heuristic_gear = settings.repair_heuristic_gear
+    @repair_heuristic_threshold = settings.repair_heuristic_threshold
+    @repair_timer = settings.repair_timer
+    UserVars.repair_timer_snap ||= Time.now
+
+    # User has option via script arg to force repair and ignore any repair timers.
+    # Handle that here. Else, exit.
+    unless args.force || should_repair?
+      echo "Skipping repair" if $debug_mode_crossing_repair
+      exit
+    end
 
-    @repair_info.each do |repairer, items|
+    @equipment_manager = EquipmentManager.new
+    @equipment_manager.wear_equipment_set?('standard')
+    @equipment_manager.empty_hands
+
+    @skip_bank = settings.sell_loot_skip_bank
+    amount, denom = settings.sell_loot_money_on_hand.split(' ')
+    @keep_copper = DRCM.convert_to_copper(amount, denom)
+    @repair_withdrawal_amount = settings.repair_withdrawal_amount
+    @hometown_data = get_data('town')[settings.hometown]
+
+    # Verify for the repair town specified, we have a bank, a leather repairer
+    # (if leather items exist), and a metal repairer (if metal items exist).
+    # Returns hash of repair items, set to variable.
+    repair_info = repair_prep(town, settings)
+
+    # Run through each item with respective metal or leather repairer, turning in items for repair.
+    repair_info.each do |repairer, items|
       echo "Repairing at #{repairer}" if $debug_mode_crossing_repair
       repair_at(repairer['name'], repairer['id'], items)
     end
 
+    # Deposit remaining coins unless we have setting to skip.
     DRCM.deposit_coins(@keep_copper, settings) unless settings.sell_loot_skip_bank
 
-    @repair_info.to_a.reverse.each do |(repairer, _items)|
+    # Handle turning in tickets to get repaired items back.
+    repair_info.to_a.reverse.each do |(repairer, _items)|
```

Comment:
❓ Can the syntax be rewritten as this:
```suggestion
    repair_info.to_a.reverse.each do |repairer, items|
```

Reply:
I don't actually know, hah! I'll try it.

Reply:
I removed the parens, but kept the underline prefix per Robert.